core/src/main/scala/pureconfig/DerivedReaders.scala

```diff
@@ -20,9 +20,26 @@ import shapeless.labelled._
 trait AllowMissingKey { self: ConfigReader[_] => }
 
 /**
+ * Trait extending [[DerivedReaders1]] that contains `ConfigReader` instances for `AnyVal`.
+ *
+ * This trait exists to give priority to the `AnyVal` derivation over the generic product derivation.
+ */
+trait DerivedReaders extends DerivedReaders1 {
```

Comment:
`<:!<` is provided by shapeless. However, I also ended up preferring the two traits solution - I gave the suggestion so that we know there is an alternative.